frontend/src/components/pages/FridgeManagement/FridgeCheckIns/index.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Button,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  HStack,
+  Input,
+  Text,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { isAfter, parse } from "date-fns";
+import React, { useState } from "react";
+import DatePicker, { DateObject } from "react-multi-date-picker";
+import { useHistory } from "react-router-dom";
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import { CreateCheckInFields } from "../../../../types/CheckInTypes";
+import ErrorMessages from "./ErrorMessages";
+
+const CreateCheckIn = () => {
+  const toast = useToast();
+  const history = useHistory();
+  const [dateRange, setDateRange] = useState<DateObject[]>([
+    new DateObject(),
+    new DateObject().add(1, "days"),
+  ]);
+  const [startTime, setStartTime] = useState<string>();
+  const [endTime, setEndTime] = useState<string>();
+  const [notes, setNotes] = useState<string>();
+  const [formErrors, setFormErrors] = useState({
+    timeRange: "",
+    dateRange: "",
+  });
+
+  const validateForm = () => {
+    const newErrors = {
+      timeRange: "",
+      dateRange: "",
+    };
+    let valid = true;
+    if (!startTime || !endTime) {
+      valid = false;
+      newErrors.timeRange = ErrorMessages.bothTimeFieldsRequired;
+    } else if (
+      isAfter(
+        parse(startTime!, "kk:mm", new Date()),
+        parse(endTime!, "kk:mm", new Date()),
+      )
+    ) {
+      valid = false;
+      newErrors.timeRange = ErrorMessages.endTimeBeforeStartTime;
+    }
+    if (!dateRange[0] || !dateRange[1]) {
+      valid = false;
+      newErrors.dateRange = ErrorMessages.bothDateFieldsRequired;
+    }
+    setFormErrors(newErrors);
+    return valid;
+  };
+
+  const onSaveClick = async () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    const checkInData: CreateCheckInFields = {
+      startDate: parse(
+        startTime!,
+        "kk:mm",
+        new Date(dateRange![0].format()),
+      ).toString(),
+      endDate: parse(
+        endTime!,
+        "kk:mm",
+        new Date(dateRange![1].format()),
+      ).toString(),
+      notes,
+      isAdmin: false,
+    };
+
+    const res = await CheckInAPIClient.createCheckIn(checkInData);
+    if (!res) {
+      toast({
+        title: "Checkin could not be created. Please try again",
+        status: "error",
+        duration: 7000,
+        isClosable: true,
+      });
+      return;
+    }
+    toast({
+      title: "Check-ins have successfully been created",
+      status: "success",
+      duration: 7000,
+      isClosable: true,
+    });
```

Comment:
nit: formatting is a bit off on mobile for toast (should be centered)
![image](https://user-images.githubusercontent.com/19617248/162593832-5f3ce0ee-15fc-418e-931d-f02681535e80.png)

Reply:
this is an issue of the text header in `ViewDonationsAndCheckins.tsx` file that the header is not being shown properly for mobile, happening for both checkins and schedules. the modal is at the bottom centre, it is just that the screen is stretched out cuz of the header text being in wrong position